src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -275,15 +278,39 @@ internal Cci.IFieldReference CreateDataField(ImmutableArray<byte> data, ushort a
                     // Use a custom type.
                     return new ExplicitSizeStruct(key.Size, key.Alignment, this, _systemValueType);
                 });
+        }
 
-            return _mappedFields.GetOrAdd((data, alignment), key =>
+        /// <summary>
+        /// Gets a field that can be used to to store data directly in an RVA field.
+        /// </summary>
+        /// <param name="data">The data for the field.</param>
+        /// <param name="alignment">
+        /// The alignment value is the necessary alignment for addresses for the underlying element type of the array.
+        /// The data is stored by using a type whose size is equal to the total size of the blob. If a built-in system
+        /// type has an appropriate size and .pack, it can be used. Otherwise, a type is generated of the same size as
+        /// the data, and that type needs its .pack set to the alignment required for the underlying data. While that
+        /// .pack value isn't required by anything else in the compiler (the compiler always aligns RVA fields at 8-byte
+        /// boundaries, which accomodates any element type that's relevant), it is necessary for IL rewriters. Such rewriters
+        /// also need to ensure an appropriate alignment is maintained for the RVA field, and while they could also simplify
+        /// by choosing a worst-case alignment as does the compiler, they may instead use the .pack value as the alignment
+        /// to use for that field, since it's an opaque blob with no other indication as to what kind of data is
+        /// stored and what alignment might be required.
+        /// </param>
+        /// <returns>The field. This may have been newly created or may be an existing field previously created for the same data and alignment.</returns>
+        internal MappedField CreateDataField(ImmutableArray<byte> data, ushort alignment)
```

Comment:
>MappedField [](http://example.com/codeflow?start=17&length=11)

Is there a specific reason for changing return type of this function?

Reply:
Yes, its MappedData.Length is accessed inside the synthesizeStaticConstructor local function.